### PR TITLE
Update `Sign`, `WAPE`, `MaxDeviation` to handle missing values

### LINKED ISSUE #525: Update `Sign`, `WAPE`, `MaxDeviation` to handle missing values
### 🚀 Task description

Currently, our `Sign`, `WAPE`, `MaxDeviation` metrics can't handle missing values and it could be fixed.

### Plan

- Update functional `sign` metric to work with missing values
  - Use `np.nanmean` like it was done for `mse`
- Update functional `wape` metric to work with missing values
  - Use `np.nansum` like it was done for `mse`
  - Make sure to return `NaN` in case of all `NaN`s
- Update functional `max_deviation` metric to work with missing values
  - Use `np.nancumsum` and `np.nanmax` like it was done for `mse`
  - Make sure that for all `NaN` the return value is `NaN`
- Update `Sign`, `WAPE`, `MaxDeviation` metrics
  - Change parent class to `MetricWithMissingHandling`
  - Add parameter `missing_mode` like it was done for `MSE`
  - Update docstring about handling missing values like for `MSE`

### Test cases

- Add tests for functional `sign`, `wape`, `max_deviation` like it was done for `mse`
- Update tests for `Sign`, `WAPE`, `MaxDeviation` to check working with missing values
  - It looks like it should be enough just to add it into parameterized tests

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by d-a-bunin**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
See #525.

## Closing issues
Closes #525

### CODE REVIEW

File: etna/metrics/functional_metrics.py [hunk 1]
```diff
@@ -355,8 +375,24 @@ def wape(y_true: ArrayLike, y_pred: ArrayLike, multioutput: str = "joint") -> Ar
         raise ValueError("Shapes of the labels must be the same")
 
     axis = _get_axis_by_multioutput(multioutput)
-
-    return np.sum(np.abs(y_true_array - y_pred_array), axis=axis) / np.sum(np.abs(y_true_array), axis=axis)  # type: ignore
+    numerator = np.nansum(np.abs(y_true_array - y_pred_array), axis=axis)
+    nans_mask = y_pred_array - y_pred_array
+    denominator = np.nansum(np.abs(y_true_array + nans_mask), axis=axis)
```
Comment (brsnw250) [inline etna/metrics/functional_metrics.py:380]:
```suggestion
    denominator = np.nansum(np.abs(y_true_array * np.isnan(y_pred_array)), axis=axis)
```

File: etna/metrics/functional_metrics.py [hunk 2]
```diff
@@ -355,8 +375,24 @@ def wape(y_true: ArrayLike, y_pred: ArrayLike, multioutput: str = "joint") -> Ar
         raise ValueError("Shapes of the labels must be the same")
 
     axis = _get_axis_by_multioutput(multioutput)
-
-    return np.sum(np.abs(y_true_array - y_pred_array), axis=axis) / np.sum(np.abs(y_true_array), axis=axis)  # type: ignore
+    numerator = np.nansum(np.abs(y_true_array - y_pred_array), axis=axis)
```
Comment (brsnw250) [inline etna/metrics/functional_metrics.py:378]: According to the documentation for versions of numpy > 1.9 `nansum` will return 0 for all empty array.  We should set it to `NaN` in such scenarios.
Seems like we don't have a test case when this behavior shows up (numerator computed from a full empty array and denominator has values).

**Review by brsnw250 — CHANGES_REQUESTED** [review]

File: tests/test_metrics/test_functional_metrics.py [hunk 3]
```diff
@@ -444,3 +444,205 @@ def test_mape_ok(y_true, y_pred, multioutput, expected):
...
+    npt.assert_allclose(result, expected)
+
+
+# TODO: исправить
+@pytest.mark.parametrize(
+    "y_true, y_pred, multioutput, expected",
+    [
+        # 1d
+        # (np.array([1.0]), np.array([1.0]), "joint", 0.0),
+        # (np.array([1.0, 2.0, 3.0]), np.array([3.0, 1.0, 2.0]), "joint", 2.0),
```
[the remark this answers is not in the document]
Comment (d-a-bunin) [reply tests/test_metrics/test_functional_metrics.py:589]: Yes, will be fixed.

File: etna/metrics/functional_metrics.py [hunk 4]
```diff
@@ -313,9 +326,14 @@ def max_deviation(y_true: ArrayLike, y_pred: ArrayLike, multioutput: str = "join
         raise ValueError("Shapes of the labels must be the same")
 
     axis = _get_axis_by_multioutput(multioutput)
-
-    prefix_error_sum = np.cumsum(y_pred_array - y_true_array, axis=axis)
-    return np.max(np.abs(prefix_error_sum), axis=axis)
+    prefix_error_sum = np.nancumsum(y_pred_array - y_true_array, axis=axis)
+    isnan = np.all(np.isnan(y_pred_array - y_true_array), axis=axis)
```
[the remark this answers is not in the document]
Comment (d-a-bunin) [reply etna/metrics/functional_metrics.py:330]: I think that this optimization isn't significant. I'll try to add it in the next commit.

**Review by brsnw250 — APPROVED** [review]

**Pull request merged**